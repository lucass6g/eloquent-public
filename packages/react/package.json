{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
