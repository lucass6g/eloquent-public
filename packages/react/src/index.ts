export {Switch, Case, Default} from './components/conditional-switch'