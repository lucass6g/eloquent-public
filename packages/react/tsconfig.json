{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
