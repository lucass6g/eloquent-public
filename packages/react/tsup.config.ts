import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})