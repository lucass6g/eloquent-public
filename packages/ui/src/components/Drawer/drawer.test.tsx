import { render, screen, fireEvent } from '@testing-library/react';
import {it, describe, expect} from "vitest"
import { Button } from '..';
import { Drawer } from '.';


describe('Drawer', () => {
    window.matchMedia = window.matchMedia || function() {
        return {
            matches: false,
            addListener: function() {},
            removeListener: function() {}
        };
    };
    it('should render the component correctly', () => {
        const component = render(<Drawer.Root>
            <Drawer.Trigger>Abrir</Drawer.Trigger>
            <Drawer.Portal>
                <Drawer.Content>
                    <Drawer.Header>
                        <Drawer.Title>Content title</Drawer.Title>
                    </Drawer.Header>
                    <Drawer.Description>Content description</Drawer.Description>
                    <Drawer.Footer>
                        <Drawer.Close asChild>
                            <Button variant="default">Cancel</Button>
                        </Drawer.Close>
                    </Drawer.Footer>
                </Drawer.Content>
                <Drawer.Overlay />
            </Drawer.Portal>
        </Drawer.Root>);

    expect(component.baseElement).toMatchSnapshot();
    })
    it('should render the drawer and handle open/close actions correctly', () => {
        const component = render(<Drawer.Root>
            <Drawer.Trigger>Abrir</Drawer.Trigger>
            <Drawer.Portal>
                <Drawer.Content>
                    <Drawer.Header>
                        <Drawer.Title>Content title</Drawer.Title>
                    </Drawer.Header>
                    <Drawer.Description>Content description</Drawer.Description>
                    <Drawer.Footer>
                        <Drawer.Close asChild>
                            <Button variant="default">Cancel</Button>
                        </Drawer.Close>
                    </Drawer.Footer>
                </Drawer.Content>
                <Drawer.Overlay />
            </Drawer.Portal>
        </Drawer.Root>);
        const trigger = screen.getByText('Abrir');
        expect(trigger).toBeInTheDocument();
        fireEvent.click(trigger);
        expect(screen.getByText('Content title')).toBeInTheDocument();
        expect(screen.getByText('Content description')).toBeInTheDocument();
        const cancelButton = screen.getByText('Cancel');
        expect(cancelButton).toBeInTheDocument();
        fireEvent.click(cancelButton);
        expect(component.baseElement).toMatchSnapshot();
    });
    it('should render the responsive variant correctly', () => {
        const component = render(<Drawer.Root>
            <Drawer.Trigger>Abrir</Drawer.Trigger>
            <Drawer.Portal>
                <Drawer.Content data-testid="content"variant="responsive">
                    <Drawer.Header>
                        <Drawer.Title>Content title</Drawer.Title>
                    </Drawer.Header>
                    <Drawer.Description>Content description</Drawer.Description>
                    <Drawer.Footer>
                        <Drawer.Close asChild>
                            <Button variant="default">Cancel</Button>
                        </Drawer.Close>
                    </Drawer.Footer>
                </Drawer.Content>
                <Drawer.Overlay />
            </Drawer.Portal>
        </Drawer.Root>);
        const trigger = screen.getByText('Abrir');
        expect(trigger).toBeInTheDocument();
        fireEvent.click(trigger);
        expect(screen.getByTestId('content')).toHaveClass('rounded-t-[10px]')
        expect(component.baseElement).toMatchSnapshot();
    });

    it('should open the drawer from right direction', () => {
        const component = render(<Drawer.Root direction='right'>
            <Drawer.Trigger>Abrir</Drawer.Trigger>
            <Drawer.Portal>
                <Drawer.Content data-testid="content"variant="responsive">
                    <Drawer.Header>
                        <Drawer.Title>Content title</Drawer.Title>
                    </Drawer.Header>
                    <Drawer.Description>Content description</Drawer.Description>
                    <Drawer.Footer>
                        <Drawer.Close asChild>
                            <Button variant="default">Cancel</Button>
                        </Drawer.Close>
                    </Drawer.Footer>
                </Drawer.Content>
                <Drawer.Overlay />
            </Drawer.Portal>
        </Drawer.Root>);
        const trigger = screen.getByText('Abrir');
        expect(trigger).toBeInTheDocument();
        fireEvent.click(trigger);
        expect(screen.getByTestId('content')).toHaveAttribute('vaul-drawer-direction', "right")
        expect(component.baseElement).toMatchSnapshot();
    });
    it('should open the drawer from left direction', () => {
        const component = render(<Drawer.Root direction='left'>
            <Drawer.Trigger>Abrir</Drawer.Trigger>
            <Drawer.Portal>
                <Drawer.Content data-testid="content"variant="responsive">
                    <Drawer.Header>
                        <Drawer.Title>Content title</Drawer.Title>
                    </Drawer.Header>
                    <Drawer.Description>Content description</Drawer.Description>
                    <Drawer.Footer>
                        <Drawer.Close asChild>
                            <Button variant="default">Cancel</Button>
                        </Drawer.Close>
                    </Drawer.Footer>
                </Drawer.Content>
                <Drawer.Overlay />
            </Drawer.Portal>
        </Drawer.Root>);
        const trigger = screen.getByText('Abrir');
        expect(trigger).toBeInTheDocument();
        fireEvent.click(trigger);
        expect(screen.getByTestId('content')).toHaveAttribute('vaul-drawer-direction', "left")
        expect(component.baseElement).toMatchSnapshot();
    });
    it('should open the drawer from top direction', () => {
        const component = render(<Drawer.Root direction='top'>
            <Drawer.Trigger>Abrir</Drawer.Trigger>
            <Drawer.Portal>
                <Drawer.Content data-testid="content"variant="responsive">
                    <Drawer.Header>
                        <Drawer.Title>Content title</Drawer.Title>
                    </Drawer.Header>
                    <Drawer.Description>Content description</Drawer.Description>
                    <Drawer.Footer>
                        <Drawer.Close asChild>
                            <Button variant="default">Cancel</Button>
                        </Drawer.Close>
                    </Drawer.Footer>
                </Drawer.Content>
                <Drawer.Overlay />
            </Drawer.Portal>
        </Drawer.Root>);
        const trigger = screen.getByText('Abrir');
        expect(trigger).toBeInTheDocument();
        fireEvent.click(trigger);
        expect(screen.getByTestId('content')).toHaveAttribute('vaul-drawer-direction', "top")
        expect(component.baseElement).toMatchSnapshot();
    });
});