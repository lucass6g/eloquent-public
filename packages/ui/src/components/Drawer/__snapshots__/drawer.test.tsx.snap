// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Drawer > should open the drawer from left direction 1`] = `
<body
  data-scroll-locked="2"
  style="right: 0px; position: fixed !important; top: 0px; left: 0px; height: auto; pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:rc:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Abrir
    </button>
  </div>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-describedby="radix-:re:"
    aria-labelledby="radix-:rd:"
    class="fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col bg-background rounded-t-[10px]"
    data-state="open"
    data-testid="content"
    id="radix-:rc:"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
    vaul-drawer=""
    vaul-drawer-direction="left"
    vaul-drawer-visible="true"
  >
    <div
      class="mx-auto mt-4 h-2 rounded-full bg-muted w-[88px]"
    />
    <div
      class="grid gap-1.5 p-4 text-center sm:text-left"
    >
      <h2
        class="mx-auto text-lg font-semibold leading-none tracking-tight"
        id="radix-:rd:"
      >
        Content title
      </h2>
    </div>
    <p
      class="mx-auto text-sm text-muted-foreground"
      id="radix-:re:"
    >
      Content description
    </p>
    <div
      class="mt-auto flex flex-col gap-2 p-4"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-base font-medium antialiased leading-4 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 disabled:bg-neutral-300 disabled:text-neutral-500 py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Drawer > should open the drawer from right direction 1`] = `
<body
  data-scroll-locked="2"
  style="right: 0px; position: fixed !important; top: 0px; left: 0px; height: auto; pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r9:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Abrir
    </button>
  </div>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-describedby="radix-:rb:"
    aria-labelledby="radix-:ra:"
    class="fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col bg-background rounded-t-[10px]"
    data-state="open"
    data-testid="content"
    id="radix-:r9:"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
    vaul-drawer=""
    vaul-drawer-direction="right"
    vaul-drawer-visible="true"
  >
    <div
      class="mx-auto mt-4 h-2 rounded-full bg-muted w-[88px]"
    />
    <div
      class="grid gap-1.5 p-4 text-center sm:text-left"
    >
      <h2
        class="mx-auto text-lg font-semibold leading-none tracking-tight"
        id="radix-:ra:"
      >
        Content title
      </h2>
    </div>
    <p
      class="mx-auto text-sm text-muted-foreground"
      id="radix-:rb:"
    >
      Content description
    </p>
    <div
      class="mt-auto flex flex-col gap-2 p-4"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-base font-medium antialiased leading-4 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 disabled:bg-neutral-300 disabled:text-neutral-500 py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Drawer > should open the drawer from top direction 1`] = `
<body
  data-scroll-locked="2"
  style="right: 0px; position: fixed !important; top: 0px; left: 0px; height: auto; pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:rf:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Abrir
    </button>
  </div>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-describedby="radix-:rh:"
    aria-labelledby="radix-:rg:"
    class="fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col bg-background rounded-t-[10px]"
    data-state="open"
    data-testid="content"
    id="radix-:rf:"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
    vaul-drawer=""
    vaul-drawer-direction="top"
    vaul-drawer-visible="true"
  >
    <div
      class="mx-auto mt-4 h-2 rounded-full bg-muted w-[88px]"
    />
    <div
      class="grid gap-1.5 p-4 text-center sm:text-left"
    >
      <h2
        class="mx-auto text-lg font-semibold leading-none tracking-tight"
        id="radix-:rg:"
      >
        Content title
      </h2>
    </div>
    <p
      class="mx-auto text-sm text-muted-foreground"
      id="radix-:rh:"
    >
      Content description
    </p>
    <div
      class="mt-auto flex flex-col gap-2 p-4"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-base font-medium antialiased leading-4 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 disabled:bg-neutral-300 disabled:text-neutral-500 py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Drawer > should render the component correctly 1`] = `
<body>
  <div>
    <button
      aria-controls="radix-:r0:"
      aria-expanded="false"
      aria-haspopup="dialog"
      data-state="closed"
      type="button"
    >
      Abrir
    </button>
  </div>
</body>
`;

exports[`Drawer > should render the drawer and handle open/close actions correctly 1`] = `
<body
  data-scroll-locked="2"
  style="position: fixed !important; top: 0px; left: 0px; right: 0px; height: auto; pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r3:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Abrir
    </button>
  </div>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0; transition: opacity 0.5s cubic-bezier(0.32,0.72,0,1);"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-describedby="radix-:r5:"
    aria-labelledby="radix-:r4:"
    class="fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col border bg-background"
    data-state="open"
    id="radix-:r3:"
    role="dialog"
    style="pointer-events: auto; transform: translate3d(0, 100%, 0); transition: transform 0.5s cubic-bezier(0.32,0.72,0,1);"
    tabindex="-1"
    vaul-drawer=""
    vaul-drawer-direction="bottom"
    vaul-drawer-visible="true"
  >
    <div
      class="mx-auto mt-4 h-2 rounded-full bg-muted w-[88px]"
    />
    <div
      class="grid gap-1.5 p-4 text-center sm:text-left"
    >
      <h2
        class="mx-auto text-lg font-semibold leading-none tracking-tight"
        id="radix-:r4:"
      >
        Content title
      </h2>
    </div>
    <p
      class="mx-auto text-sm text-muted-foreground"
      id="radix-:r5:"
    >
      Content description
    </p>
    <div
      class="mt-auto flex flex-col gap-2 p-4"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-base font-medium antialiased leading-4 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 disabled:bg-neutral-300 disabled:text-neutral-500 py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Drawer > should render the responsive variant correctly 1`] = `
<body
  data-scroll-locked="2"
  style="right: 0px; position: fixed !important; top: 0px; left: 0px; height: auto; pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r6:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      type="button"
    >
      Abrir
    </button>
  </div>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto; opacity: 0.5;"
    vaul-drawer-visible="true"
    vaul-overlay=""
    vaul-snap-points="false"
    vaul-snap-points-overlay="true"
  />
  <div
    aria-describedby="radix-:r8:"
    aria-labelledby="radix-:r7:"
    class="fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col bg-background rounded-t-[10px]"
    data-state="open"
    data-testid="content"
    id="radix-:r6:"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
    vaul-drawer=""
    vaul-drawer-direction="bottom"
    vaul-drawer-visible="true"
  >
    <div
      class="mx-auto mt-4 h-2 rounded-full bg-muted w-[88px]"
    />
    <div
      class="grid gap-1.5 p-4 text-center sm:text-left"
    >
      <h2
        class="mx-auto text-lg font-semibold leading-none tracking-tight"
        id="radix-:r7:"
      >
        Content title
      </h2>
    </div>
    <p
      class="mx-auto text-sm text-muted-foreground"
      id="radix-:r8:"
    >
      Content description
    </p>
    <div
      class="mt-auto flex flex-col gap-2 p-4"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-lg transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring focus-visible:ring-offset-1 disabled:pointer-events-none text-base font-medium antialiased leading-4 bg-primary text-primary-foreground shadow hover:bg-green-dark-400 disabled:bg-neutral-300 disabled:text-neutral-500 py-2 px-4 min-w-[152px] min-h-[24px] border-lg"
        type="button"
      >
        Cancel
      </button>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
